bare_metal_tests/TESTS/bringup/uart/test.py

```diff
@@ -0,0 +1,330 @@
+    """! This function waits for the message from the mbed device.
+         If valid message has been obtained, then it responds with
+         the same message. If invalid message has been received, then
+         it responds with 'RETRY\0'.
+         Note: First received message may be incomplete and is used for
+             synchronisation.
+             
+         @param msg expected message
+    """
+    global serial_port
+    for x in range(0, RETRY_LIMIT):
+        line = serial_read_line()
+        if(line == msg):
+            logging.info('Expected string has been received from the mbed device.')
+            serial_write_line(line)
+            return True
+        else:
+            # if the line is unexpected (corrupted) and nul terminator and send,
+            # this will trigger mbed part to resend the test message.
+            if not line.endswith('\0'):
+                line += '\0'
+            serial_write_line(line)
+        time.sleep(1)
+        
+    return False
+
+def main():
+    global serial_port
+    
+    
+    
+    sel_platform_id = None
+    muts_list = None
+    
+    # Prepare required/optional arguments.
+    parser = argparse.ArgumentParser(description='Execute UART bare metal test to check if platform is ready to use green tea.')
+    parser._action_groups.pop()
+    required = parser.add_argument_group('required arguments')
+    optional = parser.add_argument_group('optional arguments')
+    required.add_argument("-m", "--target", help="Target id")
+    optional.add_argument("-t", "--toolchain", help="Compile toolchain. Example: ARM, GCC_ARM, IAR.", default='GCC_ARM')
+    optional.add_argument("-b", "--baudrate", help="Serial port baud rate.", default=9600)
+    optional.add_argument("-c", "--clean", action='store_true', help="Perform a clean build.")
+    optional.add_argument("-v", "--verbose", action='store_true', help="Increase output verbosity.", default=0)
+    optional.add_argument("-f", "--flash_delay", help="Delay after flashing mbed board [sec].", default=1)
+    optional.add_argument("-r", "--reset_delay", help="Delay after resetting of mbed board [sec].", default=1)
+    args = parser.parse_args()
+    
+    # Set verbosity level (print transmission details or not)
+    if (args.verbose):
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+        
+    
+    # Create list of connected mbed devices.
+    mbeds = mbed_lstools.create()
+    muts_list = mbeds.list_mbeds()
+    
+    # Print platform and baud rate info.
+    logging.info('Selected platform: %s.' % (args.target))
+    logging.info('Selected serial port baud rate: %s b/s.' % (args.baudrate))
+
+    # Print list of devices with details.
+    logging.info('Available platforms: ')
+    for i, mut in enumerate(muts_list):
+        sel_str = ''
+        if(mut['platform_name'] == args.target):
+            sel_platform_id = i
+            sel_str = '  --- SELECTED ---'
```

Comment:
Why is this so complicated; why can't this `break`?

Reply:
This is done to print list of all available platforms and store the id of the selected platform for further processing. `break` can to be used since we want to show all platforms.